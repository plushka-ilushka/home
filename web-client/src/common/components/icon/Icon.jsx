import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './icon.scss';

export default function Icon({
  glyph, viewBox, className = 'icon', ...props
}) {
  return (
    <svg
      className={classNames('shape', className)}
      viewBox={viewBox}
      {...props}
    >
      <use
        xlinkHref={`#${glyph}`}
      />
    </svg>
  );
}

Icon.propTypes = {
  className: PropTypes.string,
  glyph: PropTypes.string,
  viewBox: PropTypes.string,
};

Icon.defaultProps = {
  className: '',
  glyph: '',
  viewBox: '',
};
