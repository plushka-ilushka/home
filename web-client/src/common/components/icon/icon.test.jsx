import React from 'react';
import { mount } from 'enzyme';
import Icon from './Icon';

describe('Icon Component', () => {
  it('Icon Snapshot', () => {
    expect(mount(<Icon />)).toMatchSnapshot();
  });
});
