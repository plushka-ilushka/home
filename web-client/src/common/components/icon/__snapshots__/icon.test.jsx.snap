// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component Icon Snapshot 1`] = `
<Icon
  className=""
  glyph=""
  viewBox=""
>
  <svg
    className="shape"
    viewBox=""
  >
    <use
      xlinkHref="#"
    />
  </svg>
</Icon>
`;
